app/src/main/java/space/taran/arknavigator/mvp/model/dao/ResourceExtra.kt

```diff
@@ -26,14 +27,84 @@ data class ResourceExtra(
     val value: String
 ) {
     companion object {
-        fun fromMetaExtra(id: ResourceId, extra: ResourceMetaExtra?):
-            List<ResourceExtra> =
-            extra?.data?.entries?.map { (tag, value) ->
-                ResourceExtra(
-                    resource = id,
-                    ordinal = tag.ordinal,
-                    value = value
+        fun fromMetaExtra(id: ResourceId, kind: ResourceKind?):
+            List<ResourceExtra> = when (kind) {
+            is ResourceKind.Image -> emptyList()
+            is ResourceKind.Document -> mapDocument(id, kind)
+            is ResourceKind.Video -> mapVideo(id, kind)
+            is ResourceKind.Link -> mapLink(id, kind)
+            else -> emptyList()
+        }
+
+        private fun mapDocument(
+            id: ResourceId,
+            document: ResourceKind.Document
+        ): List<ResourceExtra> {
+            val extras = mutableListOf<ResourceExtra>()
+            document.pages?.let { pages ->
+                extras.add(
+                    ResourceExtra(
+                        id,
+                        MetaExtraTag.PAGES.ordinal,
+                        pages.toString()
+                    )
+                )
+            }
+            return extras
+        }
+
+        private fun mapVideo(
+            id: ResourceId,
+            video: ResourceKind.Video
+        ): List<ResourceExtra> {
+            val valueMap = mutableMapOf<Int, String>()
+            video.height?.let {
+                valueMap[MetaExtraTag.HEIGHT.ordinal] = it.toString()
```

Comment:
Why not just generate `key to value` pairs in immutable map and map them into `ResourceExtra`s?